packages/circuit-ui/hooks/useFocusList/useFocusList.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { KeyboardEvent, useCallback, useRef } from 'react';
+
+import { uniqueId } from '../../util/id';
+import { isArrowDown, isArrowUp } from '../../util/key-codes';
+
+type FocusProps = {
+  'data-focus-list': string;
+  'onKeyDown': (event: KeyboardEvent) => void;
+};
+
+/**
+ * Enables keyboard navigation for a list of focusable elements.
+ * Spread the props returned by the hook onto each list item.
+ */
+export function useFocusList(): FocusProps {
+  const name = useRef(uniqueId());
+
+  const onKeyDown = useCallback((event: KeyboardEvent) => {
+    if (!isArrowUp(event) && !isArrowDown(event)) {
+      return;
+    }
+
+    const items = document.querySelectorAll<HTMLElement>(
+      `[data-focus-list="${name.current}"]`,
+    );
+    const currentEl = event.target as HTMLElement;
+    const currentIndex = Array.from(items).indexOf(currentEl);
+    const newIndex = isArrowUp(event)
```

Comment:
Shouldn't it also work with left/right keys? See for example the behavior for native radio buttons: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio

Reply:
I contemplated that and decided against it. It's not mentioned as a pattern in [the docs I was referencing](https://www.w3.org/WAI/tutorials/menus/application-menus/). Plus, depending on the menu, the left/right arrow keys might be needed to navigate between multiple menus.